src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1662,16 +1662,38 @@ class Simplex_tree {
  private:
   bool rec_prune_above_filtration(Siblings* sib, Filtration_value filt) {
     auto&& list = sib->members();
-    auto last = std::remove_if(list.begin(), list.end(), [this,filt](Dit_value_t& simplex) {
-        if (simplex.second.filtration() <= filt) return false;
-        if (has_children(&simplex)) rec_delete(simplex.second.children());
-        // dimension may need to be lowered
-        dimension_to_be_lowered_ = true;
-        return true;
-      });
+    bool modified = false;
+    bool emptied = false;
+    Simplex_handle last;
+
+    auto remove_children = [this, filt](Dit_value_t& simplex) {
```

Comment:
Rereading this, the name of the lambda seems a bit misleading, in particular in `if (remove_children(*sh))`. The return value of the lambda is whether we want to remove this simplex (removing the children is an implicit side effect).

Reply:
Should I name it "is_above_filtration"?

Reply:
For instance, yes. Or remove, to_remove, remove_simplex, maybe_remove, etc.

Reply:
I named it "to_remove".